12-offer-encoding.md

```diff
@@ -0,0 +1,1057 @@
+	  - MAY follow with a space and more text
+  - if it can supply more than one item for a single invoice
+    - if the minimum quantity is more than 1:
+      - MUST set that minimum in `quantity_min`
+    - if the maximum quantity is known:
+      - MUST set that maximum in `quantity_max`
+    - if neither:
+      - MUST set `quantity_min` to 1 to indicate `quantity` is supported.
+    - if both:
+      - MUST set `quantity_min` less than or equal to `quantity_max`.
+    - MUST NOT set `quantity_min` or `quantity_max` less than 1.
+  - MAY include `recurrence` to indicate offer should trigger time-spaced
+    invoices.
+  - if it includes `recurrence`:
+      - MUST set `time_unit` to 0 (seconds), 1 (days), 2 (months), 3 (years).
+      - MUST set `period` to how often (per `time-unit`) it wants to be paid.
+      - if there is a maximum number of payments:
+        - MUST include `recurrence_limit` with `max_period` set to the maximum number of payments
+        - MUST NOT set `max_period` to 0.
+      - otherwise:
+        - MUST NOT include `recurrence_limit`.
+      - if periods are always at specific time offsets:
+        - MUST include `recurrence_base`
+        - MUST set `basetime` to the initial period time in number of
+          seconds after midnight 1 January 1970
+        - if the first paid-for-period does not have to be the initial period:
+          - MUST set `start_any_period` to 1.
+        - otherwise:
+          - MUST set `start_any_period` to 0.
+      - otherwise:
+        - MUST NOT include `recurrence_base`.
+      - if payments will be accepted for the current or next period:
+        - MAY include `recurrence_paywindow`
+      - otherwise:
+        - MUST include `recurrence_paywindow`
+      - if it includes `recurrence_paywindow`:
+        - MUST set `seconds_before` to the maximum number of seconds prior to
+          a period for which it will accept payment or invoice_request for that period.
+        - MUST set `seconds_after` to the maximum number of seconds into to a
+          period for which it will accept payment or invoice_request for that period.
+        - MAY NOT enforce this for the initial period for offers without `recurrence_base`
+        - SHOULD NOT set `seconds_after` to greater than the maximum number of
+          seconds in a period.
+        - if it `amount` is specified and the node will proportionally reduce
+          the amount charged for a period payed after the start of the period:
+          - MUST set `proportional_amount` to 1 
+        - otherwise:
+          - MUST set `proportional_amount` to 0
+  - otherwise:
+    - MUST NOT include `recurrence_base`.
+    - MUST NOT include `recurrence_paywindow`.
+    - MUST NOT include `recurrence_limit`.
+  - if `send_invoice` is present:
+    - if the offer is for a partial or full refund for a previously-paid
+      invoice:
+      - SHOULD set `refunded_payment_hash` to the `payment_hash` of that
+        invoice.
```

Comment:
This feature seems completely unrelated and incompatible with the other features from offers:
- Because it has a payment hash, it means this offer is only valid for a single user and has to be generated at the request of this specific user. Since the user has to request a custom offer, she may as well directly send the invoice in her request and bypass the offer system.
- All the recurrence and quantity fields are meaningless in the context of a refund. It's the only feature that is incompatible with the others.